### PR TITLE
feat: Enable NOSF for single editions with listed price

### PR DESCRIPTION
**Pull request by jo-rs**
The type of this PR is: Feature

This PR solves [NX-3064]

### Description
This PR enables the New Offer Submission Flow on single editioned artworks with a listed price.
nosf-edition

We can continue enhancing the scope of NSFW with future work, after a clearer path forward with editions.

Based on work done by @tam-kis 
cc @artsy/negotiate-devs 

[NX-3064]: https://artsyproduct.atlassian.net/browse/NX-3064?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Order/Routes/Offer/index.tsx [hunk 1]
```diff
@@ -155,15 +155,22 @@ export class OfferRoute extends Component<OfferProps, OfferState> {
...
     const isInquiryCheckout = !artwork?.isPriceRange && !artwork?.price
-    const isEdition = artwork?.isEdition
+    const hasPrice =
+      (artwork?.listPrice?.__typename === "Money" &&
+        artwork?.listPrice?.major) ||
+      (artwork?.listPrice?.__typename === "PriceRange" &&
+        artwork?.listPrice?.maxPrice?.major &&
+        artwork?.listPrice?.minPrice?.major)
+    const showPriceOptions =
+      (artwork?.editionSets?.length ?? 0) < 2 && !!hasPrice
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/Offer/index.tsx:165]: This is just replacing `isEdition`, to see if we should be showing the "offer too low" alert on this particular flow.

File: src/v2/Apps/Order/Routes/Offer/index.tsx [hunk 2]
```diff
@@ -223,7 +230,14 @@ export class OfferRoute extends Component<OfferProps, OfferState> {
...
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/Offer/index.tsx:240]: Here's the meat and potatoes of this PR.
We are first checking if the artwork has a listed price and combining that with a check for whether the artwork has edition sets and if just one.
We'll be using this to decide whether to render the price options component or not.

Comment (tam-kis) [reply src/v2/Apps/Order/Routes/Offer/index.tsx:240]: I would at least put `artwork?.listPrice?` in a variable, or even make a minPrice and maxPrice one to cut down on the repetitiveness and make it more readable.

File: src/v2/Apps/Order/Routes/Offer/index.tsx [hunk 3]
```diff
@@ -155,15 +155,22 @@ export class OfferRoute extends Component<OfferProps, OfferState> {
     const listPriceCents = this.props.order.totalListPriceCents
     const artworkPrice = artwork?.price
     const isInquiryCheckout = !artwork?.isPriceRange && !artwork?.price
-    const isEdition = artwork?.isEdition
+    const hasPrice =
```
Comment (lean8086) [inline src/v2/Apps/Order/Routes/Offer/index.tsx:158]: Not a blocker: this seems to be duplicated, should it be a method of the class?

**Review by lean8086 — APPROVED** [review]
LGTM

**Review by tam-kis — APPROVED** [review]

**Pull request merged**